rlclientlib/extensions/onnx/src/tensor_parser.cc

```diff
@@ -0,0 +1,402 @@
+    }
+  }
+
+  template <char escape>
+  class escaped_string 
+  {
+  public:
+    struct parse_context
+    {
+      bool in_escape;
+      std::string& value;
+
+    public:
+      parse_context(std::string& target) : value(target), in_escape(false)
+      {
+      }
+
+      inline bool invoke(const char c)
+      {
+        if (in_escape || c != escape)
+        {
+          value.push_back(c);
+          in_escape = false;
+        }
+
+        if (c == escape)
+        {
+          in_escape = true;
+        }
+
+        return true;
+      }
+
+      inline bool end()
+      {
+        return true;
+      }
+    };
+
+    using parse_action = primitives::routing_action<parse_context>;
+
+    template <typename until_t>
+    using frame = primitives::parse_frame<parse_context, until_t, parse_action>;
+
+    template <char ch>
+    using until = primitives::until<parse_context, ch>;
+
+    template <bool inclusive, typename until_t>
+    static inline bool consume(const char*& c, parse_context& context)
+    {
+      return primitives::consume<inclusive, frame<until_t>, parse_context>(c, context);
+    }
+  };
+
+  using primitives::consume_exact;
+  using primitives::consume;
+  using primitives::until;
+
+  using escaped = escaped_string<BACKSLASH>;
+
+  bool parse_tensor_value(const char*& reading_head, std::vector<byte_t>& dims, std::vector<byte_t>& data, errors::error_context& error_target)
+  {
+    errors::error_context error_context = error_target.with_prefix("while parsing tensor value");
+
+    auto dims_context = base64::parse_context(dims, error_context);
+    auto data_context = base64::parse_context(data, error_context);
+
+    // " <base64 dimensions> ; <base64 data> "
+    return consume_exact<DOUBLE_QUOTE>(reading_head) &&
+      base64::consume<INCLUSIVE, base64::until<SEMICOLON>>(reading_head, dims_context) &&
+      base64::consume<INCLUSIVE, base64::until<DOUBLE_QUOTE>>(reading_head, data_context);
+  }
+
+  bool parse_tensor_name_value(const char*& reading_head, std::string& name, bytes_t& shape_bytes, bytes_t& value_bytes, errors::error_context& error_target)
+  {
+    auto name_context = escaped::parse_context(name);
+
+    // " <escaped_name> " : <tensor_value>
+    return 
+      consume_exact<DOUBLE_QUOTE>(reading_head) &&
+      (escaped::consume<INCLUSIVE, escaped::until<DOUBLE_QUOTE>>(reading_head, name_context) 
+        || // on error:
+           error_target.with_prefix("while parsing tensor name").append_error("Expected '\"'.")) &&
+      consume_exact<COLON>(reading_head) &&
+      parse_tensor_value(reading_head, shape_bytes, value_bytes, error_target);
+  }
+
+  bool parse(parser_context& context)
+  {
+    if (context._parsed)
+    {
+      return false;
+    }
+
+    const char*& reading_head = context._reading_head;
+
+    // '{' <tensor_name_value> [ ',' <tensor_name_value> ]* 
+    if (!consume_exact<OPEN_BRACE>(reading_head))
```

Comment:
It seems like improper whitespace is an error condition?

Reply:
Yes, this is correct. Right now we do not support improper (any) whitespace.